vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3295,4 +3349,225 @@ private void handleDataProviderChange(DataProvider<T, ?> dataProvider) {
         dataProviderChangeRegistration = dataProvider
                 .addDataProviderListener(event -> onDataProviderChange());
     }
+
+    /**
+     * Adds a drop listener to this component.
+     *
+     * @param listener
+     *            the listener to add, not <code>null</code>
+     * @return a handle that can be used for removing the listener
+     */
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public Registration addDropListener(
+            ComponentEventListener<GridDropEvent<T>> listener) {
+        return addListener(GridDropEvent.class,
+                (ComponentEventListener) listener);
+    }
+
+    /**
+     * Adds a drag start listener to this component.
+     *
+     * @param listener
+     *            the listener to add, not <code>null</code>
+     * @return a handle that can be used for removing the listener
+     */
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public Registration addDragStartListener(
+            ComponentEventListener<GridDragStartEvent<T>> listener) {
+        return addListener(GridDragStartEvent.class,
+                (ComponentEventListener) listener);
+    }
+
+    /**
+     * Adds a drag end listener to this component.
+     *
+     * @param listener
+     *            the listener to add, not <code>null</code>
+     * @return a handle that can be used for removing the listener
+     */
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public Registration addDragEndListener(
+            ComponentEventListener<GridDragEndEvent<T>> listener) {
+        return addListener(GridDragEndEvent.class,
+                (ComponentEventListener) listener);
+    }
+
+    /**
+     * Sets the drop mode of this drop target.
```

Comment:
Could be explicitly stated that setting a drop mode enables dropping on the grid, i.e. it starts firing drop events

Reply:
From https://github.com/vaadin/framework/blob/master/server/src/main/java/com/vaadin/ui/components/grid/GridDropTarget.java#L73 Updated.